packages/manager/src/hooks/useIsResourceRestricted.ts

```diff
@@ -0,0 +1,36 @@
+import { useEffect, useState } from 'react';
+
+import { useGrants } from 'src/queries/profile';
+
+import type { Grant, GrantLevel, GrantType, Grants } from '@linode/api-v4';
+
+/**
+ * This hook verifies if a user with restricted access can edit specific cloud resources (Linodes, NodeBalancers, Volumes, etc.).
+ * Admin-created resources may be restricted, but those created by the restricted user are editable.
+ */
+export const useIsResourceRestricted = ({
+  grantLevel = 'read_only',
+  grantType,
+  id,
+}: {
+  grantLevel?: GrantLevel;
+  grantType: GrantType;
+  grants?: Grants;
+  id: number | undefined;
+}) => {
+  const { data: grants } = useGrants();
+  const [isRestricted, setIsRestricted] = useState(false);
+
+  useEffect(() => {
+    if (grants?.[grantType]) {
+      const hasRestriction = grants[grantType].some(
+        (grant: Grant) => grant.id === id && grant.permissions === grantLevel
+      );
+      setIsRestricted(hasRestriction);
+    } else {
+      setIsRestricted(false);
+    }
+  }, [grants, id, grantType, grantLevel]);
+
+  return isRestricted;
+};
```

Comment:
I believe this can be written without needed a `useEffect` and state. I think this change works the same, but I didn't fully test it.

```suggestion
export const useIsResourceRestricted = ({
  grantLevel = 'read_only',
  grantType,
  id,
}: {
  grantLevel?: GrantLevel;
  grantType: GrantType;
  grants?: Grants;
  id: number | undefined;
}) => {
  const { data: grants } = useGrants();

  if (!grants) {
    return false;
  }

  return grants[grantType].some(
    (grant) => grant.id === id && grant.permissions === grantLevel
  );
}
```

Reply:
Yea it depends on if we want to react to grants changing or not, let me test 👍